File: src/main/java/org/dididu/controller/OmniController.java
package org.dididu.controller;

import org.dididu.service.TemplateService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;

@Controller
public class OmniController {

    private static Logger log = LoggerFactory.getLogger(OmniController.class);

    @Autowired
    private TemplateService templateService;

    @RequestMapping("/display/{userId}")
    public String greeting(@PathVariable("userId") String userId, Model model) {
        model.addAttribute("clientId", userId);

        String renderedTemplate = templateService.renderTemplateForUser(userId);
        model.addAttribute("body", renderedTemplate);

        return "omni";
    }
}

File: src/main/java/org/dididu/controller/OmniController.kt
package org.dididu.controller

import org.dididu.service.TemplateService
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping

@Controller
internal class OmniController @Autowired
constructor(private val templateService: TemplateService) {
    @RequestMapping("/{userId}/display")
    fun greeting(@PathVariable("userId") userId: String, model: Model): String {

        log.info("/display/$userId");

        model.addAttribute("clientId", userId)
        model.addAttribute("body", templateService.renderTemplateForUser(userId))
        model.addAttribute("style", templateService.getStyleForUser(userId))

        return "omni"
    }

    companion object {
        val log : Logger = LoggerFactory.getLogger(OmniController::class.java)
    }
}
